pkg/virt-api/rest/subresource.go

```diff
@@ -467,6 +508,37 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 	response.WriteHeader(http.StatusAccepted)
 }
 
+func (app *SubresourceAPIApp) findPod(namespace string, vmi *v1.VirtualMachineInstance) (string, error) {
+	fieldSelector := fields.ParseSelectorOrDie("status.phase==" + string(v12.PodRunning))
+	labelSelector, err := labels.Parse(fmt.Sprintf(v1.AppLabel + "=virt-launcher," + v1.CreatedByLabel + "=" + string(vmi.UID)))
+	if err != nil {
+		return "", err
+	}
+	selector := k8smetav1.ListOptions{FieldSelector: fieldSelector.String(), LabelSelector: labelSelector.String()}
+	podList, err := app.virtCli.CoreV1().Pods(namespace).List(selector)
+	if err != nil {
+		return "", err
+	}
+	if len(podList.Items) == 0 {
+		return "", goerror.New("connection failed. No VirtualMachineInstance pod is running")
+	} else if len(podList.Items) == 1 {
+		return podList.Items[0].ObjectMeta.Name, nil
+	} else {
+		// If we have 2 running pods, we might have a migration. Find the new pod!
+		if vmi.Status.MigrationState != nil && vmi.Status.MigrationState.Completed {
+			for _, pod := range podList.Items {
+				if pod.Name == vmi.Status.MigrationState.TargetPod {
+					return pod.Name, nil
+				}
+			}
+		} else {
+			// fallback to old behaviour
+			return podList.Items[0].ObjectMeta.Name, nil
+		}
+	}
+	return "", goerror.New("connection failed. Did not find matching VirtualMachineInstance pod")
```

Comment:
Either return a boolean, with a name of `exists`, or try to return an error which you can identify with `errors.IsNotFound`.

Reply:
@vatsalparekh, it is ok if there is no pod. Consider not returning an error in this case.

Reply:
Corrected